helix-core/src/main/java/org/apache/helix/monitoring/mbeans/dynamicMBeans/DynamicMBeanProvider.java

```diff
@@ -158,41 +155,39 @@ public synchronized void unregister() {
   }
 
   @Override
-  public Object getAttribute(String attribute)
-      throws AttributeNotFoundException, MBeanException, ReflectionException {
+  public Object getAttribute(String attribute) {
     if (SENSOR_NAME_TAG.equals(attribute)) {
       return getSensorName();
     }
 
-    if (!_attributeMap.containsKey(attribute)) {
-      return null;
-    }
+    synchronized (_lock) {
+      if (!_attributeMap.containsKey(attribute)) {
+        return null;
+      }
 
-    return _attributeMap.get(attribute).getAttributeValue(attribute);
+      return _attributeMap.get(attribute).getAttributeValue(attribute);
+    }
   }
 
   @Override
   public AttributeList getAttributes(String[] attributes) {
     AttributeList attributeList = new AttributeList();
     for (String attributeName : attributes) {
-      try {
-        Object value = getAttribute(attributeName);
-        attributeList.add(new Attribute(attributeName, value));
-      } catch (AttributeNotFoundException | MBeanException | ReflectionException ex) {
-        _logger.error("Failed to get attribute: " + attributeName, ex);
-      }
+      Object value = getAttribute(attributeName);
+      attributeList.add(new Attribute(attributeName, value));
     }
     return attributeList;
   }
 
   @Override
   public MBeanInfo getMBeanInfo() {
-    return _mBeanInfo;
+    synchronized (_lock) {
+      return _mBeanInfo;
+    }
   }
 
   @Override
-  public void setAttribute(Attribute attribute) throws AttributeNotFoundException,
-      InvalidAttributeValueException, MBeanException, ReflectionException {
```

Comment:
I know you want to keep the code clean, but this list will remind us what to do if we implement this method in the future. So I would prefer to keep them.
No strong preference. You might want more reviewers to check it : )

Reply:
In my opinion, an implemented method could throw less exceptions, but not more exceptions than the interface. If the implementation doesn't throw any exception, we don't have to put them with the implementation.
If we want to implement this method in the future, we could definitely refer to the interface.